src/r3d.imageio/r3dinput.cpp

```diff
@@ -0,0 +1,429 @@
+// Copyright Contributors to the OpenImageIO project.
+// SPDX-License-Identifier: Apache-2.0
+// https://github.com/AcademySoftwareFoundation/OpenImageIO
+
+// R3D SDK can be downloaded from the following site:
+// https://www.red.com/download/r3d-sdk
+//
+// The code has been tested with the version 8.5.1 installed in
+// /opt/R3DSDKv8_5_1 directory and setting up the variable
+// export R3DSDK_ROOT="/opt/R3DSDKv8_5_1"
+
+#include <algorithm>
+#include <cassert>
+#include <cstdio>
+
+#include <OpenImageIO/filesystem.h>
+#include <OpenImageIO/fmath.h>
+#include <OpenImageIO/imageio.h>
+#include <OpenImageIO/strutil.h>
+#include <OpenImageIO/sysutil.h>
+#include <OpenImageIO/tiffutils.h>
+
+#include <R3DSDK.h>
+#ifdef GPU
+#    ifdef CUDA
+#        include <R3DSDKCuda.h>
+#    endif  // CUDA
+#    ifdef OpenCL
+#        include <R3DSDKOpenCL.h>
+#    endif  // OpenCL
+#    ifdef Metal
+#        include <R3DSDKMetal.h>
+#    endif  // Metal
+#endif      // GPU
+#include <R3DSDKDefinitions.h>
+
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#ifdef GPU
+#    ifdef CUDA
+#        include <cuda_runtime.h>
+#    endif  // CUDA
+#endif      // GPU
+#include <vector>
+
+#include <condition_variable>
+#include <functional>
+#include <iostream>
+#include <list>
+#include <map>
+#include <mutex>
+#include <string_view>
+#include <thread>
+#include <time.h>
+#include <unistd.h>
+#include <vector>
+
+OIIO_PLUGIN_NAMESPACE_BEGIN
+
+#if 0 || !defined(NDEBUG) /* allow R3D configuration debugging */
+static bool r3d_debug = Strutil::stoi(Sysutil::getenv("OIIO_R3D_DEBUG"));
+#    define DBG(...)    \
```

Comment:
```suggestion
#    define DBG(...)   \
```

Reply:
Just modified. Wating for the build results adn leaving for today.